### PR TITLE
Split the Context into files per the TPM spec categorization

### PR DESCRIPTION
**Pull request by puiterwijk**

### CODE REVIEW

File: src/abstraction/transient.rs [hunk 1]
```diff
@@ -700,7 +700,7 @@ impl TransientKeyContextBuilder {
         context.set_sessions((Some(session), None, None));
 
         let root_key_handle = context
-            .create_primary_key(
+            .create_primary(
                 self.hierarchy,
```
Comment (Superhepper) [inline src/abstraction/transient.rs:704]: Wonderful to see the names being what you expect them to be!

**Comment by puiterwijk** [comment]
So, the one new thing I did in this was the `// Missing function: ` comments. Other than that, all of the code is literally yanked+put from `context.rs`

**Review by Superhepper — COMMENTED** [review]
This all looks great! The only thing that bugs me slightly is the name ```context_func```. To me it would have been more appropriate to either:
1. Append all files with _apis and put them directly in the context folder.
2. Or rename the ```context_func``` dir to apis.

**Comment by puiterwijk** [comment]
Sure, I'm fine with `apis`. This was just the first I came up with. In the end, it doesn't matter at all 😄

**Comment by puiterwijk** [comment]
Or maybe `crate::context::tpm_commands`? Since it's not all the APIs on `Context`.

File: src/abstraction/transient.rs (hunk 1 above)
Comment (puiterwijk) [reply src/abstraction/transient.rs]: Agreed, this has always bugged me!

**Review by ionut-arm — APPROVED** [review]
I'm ok with whichever folder names you decide to go with!

**Review by Superhepper — APPROVED** [review]
Sure tpm_commands works better.

**Pull request merged**